src/test/java/org/opensearch/neuralsearch/query/HybridQueryPostFilterIT.java

```diff
@@ -0,0 +1,743 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.query;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import lombok.SneakyThrows;
+import org.junit.Before;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.index.query.MatchNoneQueryBuilder;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.index.query.RangeQueryBuilder;
+import org.opensearch.index.query.MatchQueryBuilder;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+import static org.opensearch.neuralsearch.util.AggregationsTestUtils.getAggregations;
+import static org.opensearch.neuralsearch.util.AggregationsTestUtils.getNestedHits;
+import static org.opensearch.neuralsearch.util.AggregationsTestUtils.getAggregationValues;
+import org.opensearch.search.aggregations.AggregationBuilder;
+import org.opensearch.search.aggregations.AggregationBuilders;
+import static org.opensearch.neuralsearch.util.TestUtils.assertHitResultsFromQuery;
+
+public class HybridQueryPostFilterIT extends BaseNeuralSearchIT {
+    private static final String TEST_MULTI_DOC_INDEX_WITH_TEXT_AND_INT_MULTIPLE_SHARDS =
+        "test-hybrid-post-filter-multi-doc-index-multiple-shards";
+    private static final String TEST_MULTI_DOC_INDEX_WITH_TEXT_AND_INT_SINGLE_SHARD =
+        "test-hybrid-post-filter-multi-doc-index-single-shard";
+    private static final String SEARCH_PIPELINE = "phase-results-hybrid-post-filter-pipeline";
+    private static final String INTEGER_FIELD_1_STOCK = "stock";
+    private static final String TEXT_FIELD_1_NAME = "name";
+    private static final String KEYWORD_FIELD_2_CATEGORY = "category";
+    private static final String TEXT_FIELD_VALUE_1_DUNES = "Dunes part 2";
+    private static final String TEXT_FIELD_VALUE_2_DUNES = "Dunes part 1";
+    private static final String TEXT_FIELD_VALUE_3_MI_1 = "Mission Impossible 1";
+    private static final String TEXT_FIELD_VALUE_4_MI_2 = "Mission Impossible 2";
+    private static final String TEXT_FIELD_VALUE_5_TERMINAL = "The Terminal";
+    private static final String TEXT_FIELD_VALUE_6_AVENGERS = "Avengers";
+    private static final int INTEGER_FIELD_STOCK_1_25 = 25;
+    private static final int INTEGER_FIELD_STOCK_2_22 = 22;
+    private static final int INTEGER_FIELD_STOCK_3_256 = 256;
+    private static final int INTEGER_FIELD_STOCK_4_25 = 25;
+    private static final int INTEGER_FIELD_STOCK_5_20 = 20;
+    private static final String KEYWORD_FIELD_CATEGORY_1_DRAMA = "Drama";
+    private static final String KEYWORD_FIELD_CATEGORY_2_ACTION = "Action";
+    private static final String KEYWORD_FIELD_CATEGORY_3_SCI_FI = "Sci-fi";
+    private static final String STOCK_AVG_AGGREGATION_NAME = "avg_stock_size";
+    private static boolean setUpIsDone = false;
```

Comment:
can we put it in a separate block some if spotless permits? Other are final constants but this one keeps state.

Reply:
Can we replace state with a `BeforeClass`, similarly to [BaseAggregationsWithHybridQueryIT](https://github.com/opensearch-project/neural-search/blob/main/src/test/java/org/opensearch/neuralsearch/query/aggregation/BaseAggregationsWithHybridQueryIT.java#L84)?